server/src/api/context.go

```diff
@@ -43,15 +43,34 @@ func (s *Server) BoardCandidateContext(next http.Handler) http.Handler {
 func (s *Server) BoardParticipantContext(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		log := logger.FromRequest(r)
-
 		boardParam := chi.URLParam(r, "id")
+
 		board, err := uuid.Parse(boardParam)
 		if err != nil {
 			common.Throw(w, r, common.BadRequestError(errors.New("invalid board id")))
 			return
 		}
 
-		user := r.Context().Value("User").(uuid.UUID)
+		userValue := r.Context().Value("User")
+		if userValue == nil {
+			log.Error("User value not found in the context")
+			common.Throw(w, r, common.BadRequestError(errors.New("user not found in the context")))
+			return
+		}
+
+		user, ok := userValue.(uuid.UUID)
+		if !ok {
+			log.Error("Invalid user type in the context")
+			common.Throw(w, r, common.BadRequestError(errors.New("invalid user type in the context")))
+			return
+		}
+		if s.sessions == nil {
+			log.Error("s.sessions is nil")
+			common.Throw(w, r, common.InternalServerError)
+			return
+		}
```

Comment:
Since AuthContext is doing `uuid.Parse(userID)` and throws an error if something goes wrong, I'd expect it to be a uuid and therefore this should never throw an error.

Reply:
Since BoardParticipantContext is only called in protected routes and AuthContext is also used for all protected routes,  this check is obsolete so we can remove it